src/isa/riscv64/instr/rvv/vreg_impl.c

```diff
@@ -94,17 +99,27 @@ int get_idx(uint64_t reg, int idx, uint64_t vsew) {
   return elem_idx;
 }
 
+void isa_misalign_vreg_check(uint64_t reg, uint64_t vlmul, int needAlign) {
+  if (needAlign && vlmul < 4) {
+    if (reg % (1 << vlmul) != 0) {
+      Log("vector register group misaligned happen: reg:x%lu vlmul:0x%lx needAlign:%d", reg, vlmul, needAlign);
+      longjmp_exception(EX_II);
+    }
+  }
+}
+
 void get_vreg(uint64_t reg, int idx, rtlreg_t *dst, uint64_t vsew, uint64_t vlmul, int is_signed, int needAlign) {
   // printf("get_vreg: reg = %lu, idx = %d, vsew = %lu, vlmul = %lu, is_signed = %d, needAlign = %d\n", reg, idx, vsew, vlmul, is_signed, needAlign);
   Assert(vlmul != 4, "vlmul = 4 is reserved\n");
   Assert(vsew <= 3, "vsew should be less than 4\n");
-  if(needAlign && vlmul < 4) Assert(reg % (1 << vlmul) == 0, "vreg is not aligned\n");
+  // if(needAlign && vlmul < 4) Assert(reg % (1 << vlmul) == 0, "vreg is not aligned\n");
+  isa_misalign_vreg_check(reg, vlmul, needAlign);
   int new_reg = get_reg(reg, idx, vsew);
   int new_idx = get_idx(reg, idx, vsew);
   switch (vsew) {
-    case 0 : *dst = is_signed ? (char)vreg_b(new_reg, new_idx) : vreg_b(new_reg, new_idx); break;
-    case 1 : *dst = is_signed ? (short)vreg_s(new_reg, new_idx) : vreg_s(new_reg, new_idx); break;
-    case 2 : *dst = is_signed ? (int)vreg_i(new_reg, new_idx) : vreg_i(new_reg, new_idx); break;
+    case 0 : *dst = is_signed ? (long)(char)vreg_b(new_reg, new_idx) : vreg_b(new_reg, new_idx); break;
+    case 1 : *dst = is_signed ? (long)(short)vreg_s(new_reg, new_idx) : vreg_s(new_reg, new_idx); break;
+    case 2 : *dst = is_signed ? (long)(int)vreg_i(new_reg, new_idx) : vreg_i(new_reg, new_idx); break;
     case 3 : *dst = is_signed ? (long)vreg_l(new_reg, new_idx) : vreg_l(new_reg, new_idx); break;
   }
   // printf("get_reg: %lu idx: %d new_reg: %d new_idx: %d src: %lx\n", reg, idx, new_reg, new_idx, *dst);
```

Comment:
delete

Reply:
In practice, deleting this and running the vector results in an error

Reply:
Sorry for my wrong expression. I mean we need to replace them with width-precise type.
When we need to specify one veriable's width, we should use int64_t, int32_t, int16_t, int8_t, not just char, short, int or long, since they are not precise.